### PR TITLE
feat: allow revision pinning for rockcraft

### LINKED ISSUE #18: [Feature Request] Choose the rockcraft revision in the rockcraft-pack action
Currently the `rockcraft-pack` action supports choosing the rockcraft channel. We would like to be able to pin to a specific revision.

### PR DESCRIPTION
**Pull request by cjdcordeiro**
Closes #18 

This PR adds a new parameter to the `rockcraft-pack` action -> `revision`.
It defaults to `''`. When not empty, the action will ignore the channel and install rockcraft from the provided revision number.

### CODE REVIEW

**Comment by cjdcordeiro** [comment]
@nsklikas @sergiusens @tigarmo feel free to review

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -26,11 +26,12 @@ jobs:
       run: npm test
     - name: Try to package code
       run: npm run pack
-    
+
```
Comment (sergiusens) [inline .github/workflows/test.yml:29]: hmmm

File: rockcraft-pack/action.yml [hunk 2]
```diff
@@ -14,6 +14,12 @@ inputs:
 
       The default is 'trace'.
     default: 'trace'
+  revision:
+    description: >
+      Pin the snap revision to install.
+
+      If not provided, it defaults to whatever revision is in latest/stable.
```
Comment (sergiusens) [inline rockcraft-pack/action.yml:21]: I would say this use is potentially not recommended as there is no easy update strategy.

File: dist/rockcraft-pack-action/index.js [hunk 3]
```diff
@@ -4286,20 +4287,22 @@ class RockcraftBuilder {
...
+;// CONCATENATED MODULE: ./lib/rockcraft-pack-action.js
 // -*- mode: javascript; js-indent-level: 2 -*-
 
 
 async function run() {
     try {
         const projectRoot = core.getInput('path');
         core.info(`Building ROCK in "${projectRoot}"...`);
-        const rockcraftChannel = core.getInput('rockcraft-channel') || 'edge';
+        const rockcraftChannel = core.getInput('rockcraft-channel') || 'stable';
```
Comment (sergiusens) [inline dist/rockcraft-pack-action/index.js:4298]: good catch, maybe make this a separate PR so we roll it out faster

File: dist/rockcraft-pack-action/index.js [hunk 4]
```diff
@@ -4219,20 +4219,20 @@ async function ensureLXD() {
...
-async function ensureRockcraft(channel) {
+async function ensureRockcraft(channel, revision) {
     const haveRockcraft = await haveExecutable('/snap/bin/rockcraft');
     core.info('Installing Rockcraft...');
     await exec.exec('sudo', [
         'snap',
         haveRockcraft ? 'refresh' : 'install',
-        '--channel',
-        channel,
+        revision.length > 0 ? '--revision' : '--channel',
```
Comment (sergiusens) [inline dist/rockcraft-pack-action/index.js:4228]: this means that revision will silently mask channel, can we make it a bit more obvious that it is ignored or make it mutually exclusive in whatever schema we have?

A log message should be enough to start out.

File: .github/workflows/test.yml (hunk 1 above)
Comment (cjdcordeiro) [reply .github/workflows/test.yml:29]: cleaned

Comment (cjdcordeiro) [reply .github/workflows/test.yml:29]: cleaning

File: dist/rockcraft-pack-action/index.js (hunk 3 above)
Comment (cjdcordeiro) [reply dist/rockcraft-pack-action/index.js:4298]: well, better not cause there are people relying on this behavior to get the latest features from edge. changing this without a way for them to still be able to refer to `latest/edge` would block them

File: rockcraft-pack/action.yml (hunk 2 above)
Comment (cjdcordeiro) [reply rockcraft-pack/action.yml:21]: Can you elaborate? What's not recommended? Pinning or using the `latest/stable`?

File: dist/rockcraft-pack-action/index.js (hunk 4 above)
Comment (cjdcordeiro) [reply dist/rockcraft-pack-action/index.js:4228]: y good idea. I've added a log message in d9c1181. Making them mutually exclusive could be weird cause `channel` is not a user-defined property, but rather a fallback.

File: rockcraft-pack/action.yml (hunk 2 above)
Comment (sergiusens) [reply rockcraft-pack/action.yml:21]: I do not recall :shrug:

**Review by sergiusens — APPROVED** [review]

**Review by tigarmo — APPROVED** [review]
